lib/sequent/core/helpers/message_router.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require_relative './matchers/matchers'
+
+module Sequent
+  module Core
+    module Helpers
+      class MessageRouter
+        NO_MATCH = -> { Set.new }
```

Comment:
Should this really be a Set? Might, in theory, give unexpected results?

Reply:
I decided to use a `Set` here in order to ensure a unique collection of handlers.

The documentation of the `MessageHandler` class already states this:

> The order of which handler block is executed first is not guaranteed.

If we want guaranteed order, we could use an `OrderedSet`.

Reply:
We could also prevent users from making 'mistakes' like this by validating that provided message matchers should be unique, otherwise raise a descriptive error.

Reply:
This is not correct I think since registering two different on blocks on the same class will overwrite the first.

```ruby
on FooEvent do
  # do something
end

on FooEvent do
  # do something else
end
```

At runtime this will only execute the last one registered right? This breaks current behavior. I'd prefer to keep that the same.

Reply:
Both handlers (read: blocks) will be executed in this case.